src/vs/workbench/browser/parts/editor/textDiffEditor.ts

```diff
@@ -159,8 +159,12 @@ export class TextDiffEditor extends AbstractTextEditor<IDiffEditorViewState> imp
 			// was already asked for being readonly or not. The rationale is that
 			// a resolved model might have more specific information about being
 			// readonly or not that the input did not have.
+			const readonlyValue = resolvedDiffEditorModel.modifiedModel?.isReadonly();
```

Comment:
I feel like we could have a utility method somewhere (`editor.ts`?) that we can reuse in the 3 places.